SingularityUI/app/components/requestDetail/RequestDetailPage.jsx

```diff
@@ -45,15 +45,22 @@ class RequestDetailPage extends Component {
   }
 
   render() {
-    const { requestId } = this.props.params;
-    const { deleted } = this.props;
+    const { deleted, router, location, params } = this.props;
+    const { requestId } = params;
+    const { taskHistoryPage } =location.query;
     return (
       <div>
         <RequestHeader requestId={requestId} showBreadcrumbs={this.props.showBreadcrumbs} deleted={this.props.deleted} />
         {deleted || <RequestExpiringActions requestId={requestId} />}
         {deleted || <ActiveTasksTable requestId={requestId} />}
         {deleted || <PendingTasksTable requestId={requestId} />}
-        {deleted || <TaskHistoryTable requestId={requestId} />}
+        {deleted || (
+          <TaskHistoryTable
+            requestId={requestId}
+            onPageChange={num => router.replace(`${location.pathname}?taskHistoryPage=${num}`)}
```

Comment:
This works. You could also pass in a location object with a `query` prop for react-router. That might be a little safer in terms of ensuring that you don't override any other existing query parameters or the location hash.